File: src/structural/proxy/typescript/classes/AdminUser.ts
import { UserAddressProtocol } from './interfaces/UserAddressProtocol';
import { UserProtocol } from './interfaces/UserProtocol';
import { UserAddress } from './UserAddress';

export class AdminUser extends UserProtocol {
  async getAddresses(): Promise<UserAddressProtocol[]> {
    await new Promise((resolve) => setTimeout(resolve, 2000));
    return [new UserAddress('Street 1', 1), new UserAddress('Street 2', 2)];
  }
}

File: src/structural/proxy/python/classes/admin_user.py
import asyncio
from typing import List

from classes.interfaces.user_address_protocol import UserAddressProtocol
from classes.interfaces.user_protocol import UserProtocol
from classes.user_address import UserAddress


class AdminUser(UserProtocol):
    async def get_addresses(self) -> List[UserAddressProtocol]:
        await asyncio.sleep(2)
        return [
            UserAddress('Street 1', 1),
            UserAddress('Street 2', 2),
        ]
